lib/ruby_lsp/requests/document_link.rb

```diff
@@ -114,6 +222,25 @@ def resolve_version(uri)
 
         GEM_TO_VERSION_MAP[uri.gem_name]
       end
+
+      sig { params(node: SyntaxTree::Node, name: String).void }
+      def add_rails_document_link(node, name)
+        doc = T.must(self.class.rails_documents)[name]
+
+        return unless doc
+
+        owner = doc[:owner]
+
+        return unless RAILS_DOC_PATHS_MAP.any? do |folder, patterns|
+          @file_dir.match?(folder) && T.must(owner).match?(patterns)
+        end
+
+        @links << LanguageServer::Protocol::Interface::DocumentLink.new(
+          range: range_from_syntax_tree_node(node),
+          target: "#{RAILS_DOC_HOST}/v#{RAILTIES_VERSION}/#{doc[:path]}",
+          tooltip: "Browse the Rails documentation for: #{name}"
```

Comment:
How would you feel about adding `owner` somewhere in here? e.g. `Browse the Rails documentation for: #{owner}##{name}` or something? One thing that always confuses me about Rails is that sometimes different components can have methods with the same names, so it's not always obvious which ones you're looking at.

Reply:
Sure, but I'd use `#{name} (#{owner})` as the format instead. Because many class methods are defined as instance methods under modules like `ClassMethods` and then extended to the class ([example](https://api.rubyonrails.org/v7.0.3.1/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many)). This makes it hard to determine if we should use `.` or `#` to indicate the method type.

Reply:
IMO, we should stick with `Owner#method`. It's a convention used everywhere in Ruby and the owner being at the back is a bit odd.